hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpGetCommandProcessor.java

```diff
@@ -71,14 +77,40 @@ private void handleHealthcheck(HttpGetCommand command) {
         boolean clusterSafe = memberStateSafe && !partitionService.hasOnGoingMigration();
         long migrationQueueSize = partitionService.getMigrationQueueSize();
 
-        StringBuilder res = new StringBuilder();
-        res.append("Hazelcast::NodeState=").append(nodeState).append("\n");
-        res.append("Hazelcast::ClusterState=").append(clusterState).append("\n");
-        res.append("Hazelcast::ClusterSafe=").append(Boolean.toString(clusterSafe)
-                .toUpperCase(StringUtil.LOCALE_INTERNAL)).append("\n");
-        res.append("Hazelcast::MigrationQueueSize=").append(migrationQueueSize).append("\n");
-        res.append("Hazelcast::ClusterSize=").append(clusterSize).append("\n");
-        command.setResponse(MIME_TEXT_PLAIN, stringToBytes(res.toString()));
+        String healthParameter = uri.substring(URI_HEALTH_URL.length());
+        if (healthParameter.equals(HEALTH_PATH_PARAM_NODE_STATE)) {
+            if (NodeState.SHUT_DOWN.equals(nodeState)) {
+                command.setResponse(HttpCommand.RES_503);
+            } else {
+                command.setResponse(null, stringToBytes(nodeState.toString()));
```

Comment:
Can you check if it is meaningful to return 200 in case of `ACTIVE` nodestate?

Reply:
I think that both `ACTIVE` and `PASSIVE` mean that the node is UP, so 200 sounds reasonable. It's a little similar to what Spring Actuator does [1], they return 200 for both UNKNOWN and UP (with the message). 

Also, I don't see any other code that could be returned to distinguish them, plus we don't need it for Kubernetes/OpenShift. 

[1] https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#_writing_custom_healthindicators

Reply:
Hello, we're having some troubles while using Hazelcast as a StatefulSet in k8s. Sometimes Hazelcast isn't ready while we see that /hazelcast/health/ready endpoint responds Http Status=200. It just happens on the first deployment but it affects our integration tests since they're deployed every time we want to run the tests. Do you think it's related to Node being PASSIVE? Do you think might it resolve the issue if I add this check to the healthProbe? Thanks a lot!

Reply:
Hi @enesunal ,
Are you using [Hazelcast Helm Charts](https://github.com/hazelcast/charts/) or you have your own deployment configs?

As you see, hazelcast helm chart uses [node-state](https://github.com/hazelcast/charts/blob/master/stable/hazelcast/templates/statefulset.yaml#L72) instead of `/ready`.

Reply:
Hi @mesutcelik; woow that was such a quick response. Yes, we're using our own deployment configs. In k8s, being ready means being able to accept traffic. So, I guess it would be better to have it in the same way that almost all the applications have. But I'll try to check helm charts and figure out how it's done. Thanks a lot!!